app/models/post_creator.rb

```diff
@@ -0,0 +1,24 @@
+class PostCreator
+  class PostCreateError < RuntimeError; end
+
+  public
+  def self.create_post(message, amount, sender, receivers, team)
+    post = Post.new(
+        message: message,
+        amount: amount,
+        sender: sender,
+        receivers: receivers,
+        team: team,
+        kudos_meter: team.active_kudos_meter
+    )
+
+    if post.save
+      PostMailer.new_post(post)
+      GoalReacher.check!(post.team)
```

Comment:
I think we should make the Goal check transactional because if that fails it might introduce a weird state in the application. 
The slack announcement and mail announcement however do not influence the state of the application and, in my opinion, can be ignored. For example we do not want the creation of a post to fail when the smtp settings for the mail client are wrong.

Reply:
:+1: sounds like a good idea.